provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -839,6 +839,9 @@ func (c *Client) buildConfigSpec(
 			Reservation: &cpuPower,
 		}
 	}
+	if args.Constraints.HasNestedVirtualization() {
+		spec.NestedHVEnabled = types.NewBool(args.Constraints.CpuPower)
```

Comment:
I'm guessing this is just a typo, and you actually meant something other than casting CpuPower (an *uint64) to a Bool. Probably you wanted types.NewBool(args.Constraints.NestedVirtualization) ?